ui/apps/platform/src/Containers/Policies/Wizard/Step3/BooleanPolicyLogicSection.tsx

```diff
@@ -21,23 +14,19 @@ type BooleanPolicyLogicSectionProps = {
 };
 
 function BooleanPolicyLogicSection({ readOnly = false }: BooleanPolicyLogicSectionProps) {
-    const [descriptor, setDescriptor] = React.useState<Descriptor[]>([]);
     const { values } = useFormikContext<Policy>();
+    const { isFeatureFlagEnabled } = useFeatureFlags();
 
-    const isImageSigningEnabled = useFeatureFlagEnabled('ROX_VERIFY_IMAGE_SIGNATURE');
-    const isNetworkPolicyFieldsEnabled = useFeatureFlagEnabled('ROX_NETPOL_FIELDS');
-    React.useEffect(() => {
-        if (values.eventSource === 'AUDIT_LOG_EVENT') {
-            setDescriptor(auditLogDescriptor);
-        } else {
-            setDescriptor([
-                ...policyConfigurationDescriptor,
-                ...networkDetectionDescriptor,
-                ...(isImageSigningEnabled ? [imageSigningCriteriaDescriptor] : []),
-                ...(isNetworkPolicyFieldsEnabled ? networkPolicyFieldDescriptors : []),
-            ]);
+    const unfilteredDescriptors =
+        values.eventSource === 'AUDIT_LOG_EVENT'
+            ? auditLogDescriptor
+            : policyConfigurationDescriptor;
+    const descriptors = unfilteredDescriptors.filter((unfilteredDescriptor) => {
+        if (typeof unfilteredDescriptor.featureFlagDependency === 'string') {
```

Comment:
why is this type check necessary? what's the benefit of doing it this way vs just checking if the property exists on the descriptor?

Reply:
Good eyes. The type check is habit to distinguish falsey values like false or zero or empty string from undefined or property does not exist, although indeed a feature flag will not be an empty string.

Reply:
However, simpler condition might have been better, because TypeScript got confused about string and string union type for property from object destructuring instead of property with dot notation.